### PR TITLE
Add minimal estimators docs

### PR DESCRIPTION
**Pull request by adonath**
**Description**

This pull request adds some minimal content to the `gammapy.estimators` sub-package, explaining the general approach and  quantities involved.

### CODE REVIEW

File: docs/estimators/index.rst [hunk 1]
```diff
@@ -11,12 +11,76 @@ estimators - High level estimators
...
+flux points, flux profiles and flux light curves. All estimators
+feature a common API and allow to estimate fluxes in bands of reconstructed
+energy.
+
+The core of any estimator algorithm is hypothesis testing: a reference
+model or counts excess is tested against a null hypothesis. From the
+best fit reference model a flux is derived and a corresponding sqrt(TS)
+value from the difference in fit statistics to the null hypothesis,
+assuming one degree of freedom (TODO: link to stats). In this case
+sqrt(TS) represents an approximation of the classical significance.
```
Comment (registerrier) [inline docs/estimators/index.rst:24]: We need a proper discussion of significance vs sqrt(TS) somewhere.

**Review by registerrier — APPROVED** [review]
Thanks @adonath !

We might discuss the two different types of estimators, since some perform real forward folding and some others don't (ExcessEstimators).

File: docs/estimators/index.rst (hunk 1 above)
Comment (adonath) [reply docs/estimators/index.rst:24]: For now I added a reference to `gammapy.stats`. I think this should be the single place where we explain thi sin more detail...

Comment (adonath) [reply docs/estimators/index.rst]: In the stats section I added the "recipe" how derive the significance from delta_ts in general. It's not yet in a clear state, but at least the information is somewhere...we can still further improve this of course.

File: docs/estimators/index.rst [hunk 2]
```diff
@@ -11,12 +11,76 @@ estimators - High level estimators
...
+
```
[the remark this answers is not in the document]
Comment (adonath) [reply docs/estimators/index.rst:38]: I added a bullet list explaining the two options, but didn't give the overview on the estimators yet...

**Comment by adonath** [comment]
Thanks @registerrier and @bkhelifi I'll merge this PR now, so that we can improve independently the `gammapy.stats` and `gammapy.estimators` pages in follow up PRs. I'll trigger a new docs build and then you can review again on basis of the dev docs, which is maybe easier...

**Pull request merged**